src/main/java/dev/binclub/javaception/classfile/ClassFileParser.java

```diff
@@ -1,4 +1,257 @@
+		case "SourceFile":
+			return new SourceFileAttribute(attributeNameIndex, attributeLength, dis);
+		default:
+			System.out.println("skipping attrib type " + name);
+			dis.skipBytes(attributeLength);
+			break;
+		}
+		return null;
+	}
+
+	public void getInterfaces() throws Throwable {
+		interfaces = new ClassInfoCP[interfacesCount];
+		for (int i = 0; i < interfacesCount; i++) {
+			interfaces[i] = (ClassInfoCP) constantPool[dis.readUnsignedShort() - 1];
+		}
+	}
+
+	public void parseConstantPool() throws Throwable {
+
+		for (int i = 0; i < constantPoolCount - 1; i++) {
+			int tag = dis.readUnsignedByte();
+			ConstantTypes type = types[tag];
+			switch (type) {
+			case CLASS:
+				constantPool[i] = new ClassInfo(dis.readUnsignedShort());
+				break;
+			case DOUBLE:
+				constantPool[i] = dis.readDouble();
+				break;
+			case FIELDREF:
+				int classIndex = dis.readUnsignedShort();
+				int nameAndTypeIndex = dis.readUnsignedShort();
+				constantPool[i] = new RefInfo(classIndex, nameAndTypeIndex);
+				break;
+			case FLOAT:
+				constantPool[i] = dis.readFloat();
+				break;
+			case INTEGER:
+				constantPool[i] = dis.readInt();
+				break;
+			case INTERFACEMETHODREF:
+				classIndex = dis.readUnsignedShort();
+				nameAndTypeIndex = dis.readUnsignedShort();
+				constantPool[i] = new RefInfo(classIndex, nameAndTypeIndex);
+				break;
+			case INVOKEDYNAMIC:
+				int bootstrapMethodAttrIndex = dis.readUnsignedShort();
+				nameAndTypeIndex = dis.readUnsignedShort();
+				constantPool[i] = new InvokeDynamicInfo(bootstrapMethodAttrIndex, nameAndTypeIndex);
+				break;
+			case LONG:
+				constantPool[i] = dis.readLong();
+				break;
+			case METHODHANDLE:
+				int referenceKind = dis.readUnsignedByte();
+				int referenceIndex = dis.readUnsignedShort();
+				constantPool[i] = new MethodHandleInfo(referenceKind, referenceIndex);
+				break;
+			case METHODREF:
+				classIndex = dis.readUnsignedShort();
+				nameAndTypeIndex = dis.readUnsignedShort();
+				constantPool[i] = new RefInfo(classIndex, nameAndTypeIndex);
+				break;
+			case METHODTYPE:
+				int descriptorIndex = dis.readUnsignedShort();
+				constantPool[i] = new MethodTypeInfo(descriptorIndex);
+				break;
+			case NAMEANDTYPE:
+				int nameIndex = dis.readUnsignedShort();
+				descriptorIndex = dis.readUnsignedShort();
+				constantPool[i] = new NameAndTypeInfo(nameIndex, descriptorIndex);
+				break;
+			case STRING:
+				int stringIndex = dis.readUnsignedShort();
+				constantPool[i] = new UnresolvedString(stringIndex);
+				break;
+			case UTF8:
+				constantPool[i] = dis.readUTF();
+				break;
+			case DYNAMIC:
+				bootstrapMethodAttrIndex = dis.readUnsignedShort();
+				nameAndTypeIndex = dis.readUnsignedShort();
+				constantPool[i] = new DynamicInfo(bootstrapMethodAttrIndex, nameAndTypeIndex);
+				break;
+			default:
+				System.out.println("Unsupported constant type : " + tag);
```

Comment:
This should be a `ClassFormatError`. 

In the future it might also be a good idea to make sure that constant pool entry types only appear in class versions that are valid, for example no Dynamic constant pool entry in a java 6 classfile.